pico_flash should depend on hardware_exception

Hi there, I think pico_flash should depend on hardware_exception? It unconditionally includes hardware_exception, and if I try to compile a file that depends on pico_flash without editing the SDK, I get this compile error:

```
[build] /home/jaguilar/projects/pico-sdk/src/rp2_common/pico_flash/flash.c:8:10: fatal error: hardware/exception.h: No such file or directory
[build]     8 | #include "hardware/exception.h"
[build]       |          ^~~~~~~~~~~~~~~~~~~~~~
```

If I add hardware_exception to the dependency list of pico_flash then everything works as expected.

--- a/src/rp2_common/pico_flash/flash.c
+++ b/src/rp2_common/pico_flash/flash.c
@@ -5,7 +5,6 @@
  */
 
 #include "pico/flash.h"
-#include "hardware/exception.h"
 #include "hardware/sync.h"
 #if PICO_FLASH_SAFE_EXECUTE_PICO_SUPPORT_MULTICORE_LOCKOUT
 #include "pico/multicore.h"

--- a/src/rp2_common/pico_flash/include/pico/flash.h
+++ b/src/rp2_common/pico_flash/include/pico/flash.h
@@ -9,9 +9,6 @@
 
 #include "pico.h"
 
-#include "hardware/flash.h"
-#include "pico/time.h"
-
 /** \file pico/flash.h
  *  \defgroup pico_flash pico_flash
  *

